test/pylib/scylla_server.py

```diff
@@ -0,0 +1,307 @@
+
+    @property
+    def host(self) -> str:
+        return self.cfg["host"]
+
+    def find_scylla_executable(self) -> None:
+        if not os.access(self.exe, os.X_OK):
+            raise RuntimeError("{} is not executable", self.exe)
+
+    async def install(self) -> None:
+        """Create a working directory with all subdirectories, initialize
+        a configuration file."""
+
+        self.find_scylla_executable()
+
+        # Scylla assumes all instances of a cluster use the same port,
+        # so each instance needs an own IP address.
+        self.cfg["host"] = await self.host_registry.lease_host()
+        self.cfg["seeds"] = self.cfg.get("seeds") or self.cfg["host"]
+        # Use the last part in host IP 127.151.3.27 -> 27
+        # There can be no duplicates within the same test run
+        # thanks to how host registry registers subnets, and
+        # different runs use different vardirs.
+        self.shortname = pathlib.Path("scylla-" + self.cfg["host"].split(".")[-1])
+        self.cfg["workdir"] = self.vardir / self.shortname
+
+        logging.info("installing Scylla server in %s...", self.cfg["workdir"])
+
+        self.log_file_name = self.vardir / self.shortname.with_suffix(".log")
+
+        self.config_file_name = self.cfg["workdir"] / "conf/scylla.yaml"
+
+        # Use tmpdir (likely tmpfs) to speed up scylla start up, but create a
+        # link to it to keep everything in one place under testlog/
+        self.tmpdir = pathlib.Path(os.getenv('TMPDIR', '/tmp'))
+        self.tmpdir = self.tmpdir / ('scylla-'+self.cfg["host"])
+
+        # Cleanup any remains of the previously running server in this path
+        shutil.rmtree(self.tmpdir, ignore_errors=True)
+        pathlib.Path(self.cfg["workdir"]).unlink(missing_ok=True)
+
+        pathlib.Path(self.tmpdir).mkdir(parents=True, exist_ok=True)
+        pathlib.Path(self.cfg["workdir"]).symlink_to(self.tmpdir)
+        pathlib.Path(self.config_file_name).parent.mkdir(parents=True, exist_ok=True)
+        # Create a configuration file.
+        with open(self.config_file_name, 'w') as config_file:
+            config_file.write(SCYLLA_CONF_TEMPLATE.format(**self.cfg))
+
+        self.log_file = open(self.log_file_name, "wb")
+
+    def find_log_file_pattern(self, fil, pattern_re) -> bool:
+        for line in fil.readlines():
+            if pattern_re.match(line):
+                return True
+        return False
+
+    async def cql_is_up(self) -> bool:
+        """Test that CQL is serving (a check we use at start up)."""
+        caslog = logging.getLogger('cassandra')
+        oldlevel = caslog.getEffectiveLevel()
+        # Be quiet about connection failures.
+        caslog.setLevel('CRITICAL')
+        auth = PlainTextAuthProvider(username='cassandra', password='cassandra')
+        try:
+            with Cluster(contact_points=[self.cfg["host"]], auth_provider=auth) as cluster:
+                with cluster.connect():
```

Comment:
Are we fine with this blocking? (not sure how the async code interacts with driver code)

Reply:
It is blocking. For now we're fine with it, yes.